mass_mailing_switzerland/models/res_partner_category.py

```diff
@@ -17,14 +17,22 @@ class ResPartnerCategory(models.Model):
     def update_partner_tags(self):
         # Update mailing contacts and mailchimp when tags are recomputed.
         for tag in self:
+            old_partners = set(tag.mapped("partner_ids").ids)
             tag.mapped("partner_ids.mailing_contact_ids").write({
                 "tag_ids": [(3, tag.id)]
             })
-        super().update_partner_tags()
-        for tag in self:
+            super(ResPartnerCategory, tag).update_partner_tags()
+            new_partners = set(tag.mapped("partner_ids").ids)
             tag.mapped("partner_ids.mailing_contact_ids").write({
                 "tag_ids": [(4, tag.id)]
             })
-        self.env["mail.mass_mailing.contact"] \
-            .with_delay().update_all_merge_fields_job(self.mapped("partner_ids").ids)
+            for partner_id in old_partners ^ new_partners:
+                queue_job = self.env["queue.job"].search([
+                    ("channel", "=", "root.mass_mailing_switzerland.update_partner_mailchimp"),
+                    ("state", "!=", "done"),
+                ])
+                if len(queue_job) and partner_id in queue_job.mapped("args"):
+                    continue
+                self.env["mail.mass_mailing.contact"] \
+                    .with_delay().update_partner_merge_fields_job(partner_id)
```

Comment:
Here we could as well simply call the job functions on the right recordset (`old_partners ^ new_partners`) if we move the logic of searching for existing jobs inside the `update_partner_merge_fields_job` function.